streams/src/main/java/org/apache/kafka/streams/StreamConfig.java

```diff
@@ -121,30 +116,58 @@
     private static final String SYSTEM_TEMP_DIRECTORY = System.getProperty("java.io.tmpdir");
 
     static {
-        CONFIG = new ConfigDef().define(JOB_ID_CONFIG,
+        CONFIG = new ConfigDef().define(JOB_ID_CONFIG,      // required with no default value
                                         Type.STRING,
-                                        "",
-                                        Importance.MEDIUM,
-                                        StreamingConfig.JOB_ID_DOC)
+                                        Importance.HIGH,
+                                        StreamConfig.JOB_ID_DOC)
+                                .define(BOOTSTRAP_SERVERS_CONFIG,       // required with no default value
+                                        Type.STRING,
+                                        Importance.HIGH,
+                                        CommonClientConfigs.BOOSTRAP_SERVERS_DOC)
                                 .define(CLIENT_ID_CONFIG,
```

Comment:
Out of scope but when would it make sense to have Client ID defined for streams? I mean, we'll generate a lot of clients when the job runs, right?